### PR TITLE
Add load persistence_guid from UserData [11689]

### PR DESCRIPTION
**Pull request by richiware**
Feature for distributing persistence guid through participant user_data when EDP static is used

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

File: src/cpp/rtps/builtin/discovery/endpoint/EDPStatic.cpp [hunk 1]
```diff
@@ -214,9 +214,35 @@ bool EDPStatic::removeLocalWriter(
...
+        persistence_guid.guidPrefix.value[11] = pdata.m_userData.at(14);
+        persistence_guid.entityId.value[0] = pdata.m_userData.at(15);
+        persistence_guid.entityId.value[1] = pdata.m_userData.at(16);
+        persistence_guid.entityId.value[2] = pdata.m_userData.at(17);
+    }
+
     for (ParameterPropertyList_t::const_iterator pit = pdata.m_properties.begin();
             pit != pdata.m_properties.end(); ++pit)
     {
+        persistence_guid.entityId.value[3] = 0;
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/discovery/endpoint/EDPStatic.cpp:244]: value[3] is the `entityKind`, so I think the user id should be assigned to either index 0 or 2

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
1. Shouldn't we have a test for this feature?
2. Fix linters

File: src/cpp/rtps/builtin/discovery/endpoint/EDPStatic.cpp (hunk 1 above)
Comment (richiware) [reply src/cpp/rtps/builtin/discovery/endpoint/EDPStatic.cpp:244]: This is actually not an option because will break interoperability

Comment (MiguelCompany) [reply src/cpp/rtps/builtin/discovery/endpoint/EDPStatic.cpp:244]: As talked offline, the persistence_guid is always set by the user, and doesn't need to follow the standard GUID rules.

**Comment by richiware** [comment]
@MiguelCompany Applied suggestions

**Review by MiguelCompany — DISMISSED** [review]
LGTM

**Review by MiguelCompany — APPROVED** [review]
LGTM. Failures do seem unrelated.

**Pull request merged**